File: Python/Algorithm/Algorithm I/167. Two Sum II - Input Array Is Sorted.py
from typing import List


class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        """
        Find two numbers in a sorted array that add up to a specific target.

        Args:
            numbers (List[int]): A list of integers.
            target (int): The target sum.

        Returns:
            List[int]: A list containing the indices of the two numbers that add up to the target.
        """
        NUMBERS_LENGTH = len(numbers)

        leftPointer = 0
        rightPointer = NUMBERS_LENGTH - 1

        while leftPointer < rightPointer:
            currentSum = numbers[leftPointer] + numbers[rightPointer]

            if currentSum == target:
                return [leftPointer + 1, rightPointer + 1]

            elif currentSum < target:
                leftPointer += 1

            else:
                rightPointer -= 1

File: Java/Algorithm/Algorithm I/167. Two Sum II - Input Array Is Sorted.java
class Solution {

  /**
   * Finds two numbers in a sorted array that add up to a given target sum.
   *
   * @param numbers An array of integers sorted in ascending order.
   * @param target  The target sum to find.
   * @return An array containing the indices (1-based) of the two numbers that add up to the target sum.
   */
  public int[] twoSum(int[] numbers, int target) {
    int NUMBERS_LENGTH = numbers.length;

    int leftPointer = 0;
    int rightPointer = NUMBERS_LENGTH - 1;

    while (leftPointer < rightPointer) {
      int sumValue = numbers[leftPointer] + numbers[rightPointer];

      if (sumValue == target) break;
      else if (sumValue > target) rightPointer--; 
      else leftPointer++;
    }

    return new int[] { leftPointer + 1, rightPointer + 1 };
  }
}
